### ISSUE TITLE
Updated field names in JSON file ?

### ISSUE THREAD

**Issue by vkaelin** [meraki-analytics/role-identification#8]
Hello! 

It looks like the json file has been updated with new role names ? 
http://cdn.merakianalytics.com/riot/lol/resources/latest/en-US/championrates.json

E.g. it has "ADC", "SUPPORT" where the script uses "BOTTOM", "UTILITY". 

Is it something that should be changed ?

Best,
Valentin

**Comment by vkaelin** [comment]
And I think there is 2 missing champions in the cdn file: http://cdn.merakianalytics.com/riot/lol/resources/latest/en-US/championrates.json

It only has 146 champions. I think it's Heimerdinger (74) and Taliyah (163).

**Comment by jjmaldonis** [comment]
In response to your first comment, that file is not intended to be used externally. Is there a reason you are accessing it? (I did update the json to reflect the role names used in our code, however, and updated Cassiopeia to go with it.)

For your second comment, it looks like those exclusions are intended - just like Lee Sin middle is excluded. The play rates for these champs+roles (and all roles for Heimer and Taliyah evidently) are too low to be relevant.

What are you using this file for?

**Comment by vkaelin** [comment]
It's strange that not all champions are in the json because your script crashes when you try to get a team with Heimer for example.

I'm using the json file because I made a fork of the script in Node.js.
Sadly the results are not so great, when I get one "exotic" pick (e.g. Brand mid has 0% playRate in the file).
I think I'll try to use your RoleML algorithm by calling the python script from my Node.js API with something like Python-Shell.

Just curious, how do you create the playRate by position file ? How many matches do you scan ?

Thanks!

**Comment by jjmaldonis** [comment]
I added Heimer to the current json file so that it no longer crashes.

I also tested a few games with Brand in the team composition, and it is working as I would expect it to given the inputs. Canisback's roleML is certainly more accurate because it includes post-game data. It is much harder to implement, train, keep up to date, and get data for, so that is the trade off: more/less accuracy for more/less work+time.

This data is compiled from millions of matches in most regions, so not just NA.

If you want to use this code in another language, I've been thinking about allowing the world to hit an API endpoint that will return these results. You would make a POST request to a URL with the team composition you want the role for, and the response would be the role assignments. I have all of this set up already but the person who said they would test it isn't responsive, so I'm looking for someone else. I want to make sure it 1) works well and 2) isn't expensive. If you want to give this a try (and it should be really simple to implement), DM me on discord and I'll give you the URL and POST body syntax. This will allow you to use our code in any language, at the expense of an http request.
